siatest/consensus/consts.go

```diff
@@ -0,0 +1,19 @@
+package consensus
+
+import (
+	"os"
+	"testing"
+
+	"github.com/NebulousLabs/Sia/siatest"
+)
+
+// consensusTestDir creates a temporary testing directory for a consensus. This
+// should only every be called once per test. Otherwise it will delete the
+// directory again.
+func consensusTestDir(t *testing.T) string {
```

Comment:
We generally try not to pass around the `testing.T` object, because certain methods of that object only work if you are using it in the main goroutine, and we've run into issues with it before. Better would be to pass around the result of calling `t.Name()` since that's the only thing you need ti for.

Reply:
Ok. My thought process was that there is no reason for anyone to ever be able to pass a custom name since it would violate the naming convention.

Reply:
IIRC the main annoyance we had with passing around `testing.T` is that it made failing tests harder to understand. If the helper function failed, you'd only see the name+line of the helper function, not the test that called it, so it could be difficult to track down the precise cause of the failure.
However, Go fixed this by adding the `t.Helper()` method, which causes the helper function to be omitted from the error trace. 

As for the main goroutine, the only issue I'm aware of is that `t.Fatal` only works from the main goroutine. But that's an important concern, because most of the appeal of passing around `testing.T` is that you can hide all the `t.Error` and `t.Fatal` calls in helper functions. So you'd have to be careful to avoid calling helper functions in goroutines.

My opinion on this has shifted over the years, but currently I think it's best to stick with plain old error return values. Go tests should strive to look like normal Go code, which means eschewing magic. So with regard to `t.Name()`, I agree with David that it should be passed as a string, not a `testing.T`. It makes it clear that the function only uses the name and won't cause the test to fail or something, and it also allows for custom names in the rare event that we need one (and most likely, we will).